includes/polling/os/radlan.inc.php

```diff
@@ -1,10 +1,21 @@
 <?php
 
-$version  = snmp_get($device, 'rndBrgVersion.0', '-Ovq', 'RADLAN-MIB');
-$hardware = str_replace('ATI', 'Allied Telesis', $device['sysDescr']);
+// sysDescr.0 = STRING: "ATI AT-8000S"
+// sysDescr.0 = STRING: 48-port 10/100/1000 Ethernet Switch 
+// sysDescr.0 = STRING: 24-port 10/100/1000 Ethernet Switch with PoE
+
+$hardware = snmp_getnext($device, '.1.3.6.1.2.1.47.1.1.1.1.2.64', '-OsvQU', '+ENTITY-MIB');
+$version  = snmp_get($device, '.1.3.6.1.4.1.89.2.4.0', '-OsvQU', '+SNMPv2-SMI');
+$serial   = snmp_get($device, 'ENTITY-MIB::entPhysicalSerialNum.64', '-OsvQU', '+ENTITY-MIB');
 
+//legacy radlan. Not sure which devices use this. 
+
+if (!$hardware && !$version && !$serial) {
+$version  = snmp_get($device, 'rndBrgVersion.0', '-Ovq', 'RADLAN-MIB');
```

Comment:
As you now have two snmp queries, move them to `snmp_get_multi_oid`

Reply:
As I've been fixing this up, I did some more analysis on the OIDs.

I currently have:

```
$hardware = snmp_getnext($device, 'entPhysicalDescr.64', '-OsvQU', 'ENTITY-MIB');
$version  = snmp_get($device, 'rndBrgVersion.0', '-OsvQU', 'RADLAN-MIB');
$serial   = snmp_get($device, 'entPhysicalSerialNum.64', '-OsvQU', 'ENTITY-MIB');

//legacy radlan. Not sure which devices use this. 

if (!$hardware && !$version && !$serial) {
$version  = snmp_get($device, 'rndBrgVersion.0', '-Ovq', 'RADLAN-MIB');
$hardware = str_replace('ATI', '', $device['sysDescr']);
$features = snmp_get($device, 'rndBaseBootVersion.00', '-Ovq', 'RADLAN-MIB');
}
```

Now that snmp_getnext exists in librenms, is possible that "$hardware = str_replace('ATI', '', $device['sysDescr']);" is no longer needed.  

I've tested on two different models:

```
librenms@librenmsdev:~$ snmpgetnext -v2c -c librenms -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/radlan 10.37.105.92 SysDesc
MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/radlan
48-port 10/100/1000 Ethernet Switch
librenms@librenmsdev:~$ snmpgetnext -v2c -c librenms -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/radlan 10.37.105.91 SysDesc
ATI AT-8000S

As you can see, both models while running the same OS have vastly different System Descriptions. However, both provide correct info when:

librenms@librenmsdev:~$ snmpgetnext -v2c -c librenms -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/radlan 10.37.105.91 entPhysicalDescr.64
AT-8000S/24   
librenms@librenmsdev:~$ snmpgetnext -v2c -c librenms -OsvQU -M /opt/librenms/mibs:/opt/librenms/mibs/radlan 10.37.105.92 entPhysicalDescr.64
AT-8000GS/48

Therefore, I'm just going to use the snmpgetnext instead.

After going through above, this is all a lot more simple:

```
<?php

// sysDescr.0 = STRING: "ATI AT-8000S"
// sysDescr.0 = STRING: 48-port 10/100/1000 Ethernet Switch 
// sysDescr.0 = STRING: 24-port 10/100/1000 Ethernet Switch with PoE

$hardware = snmp_getnext($device, 'entPhysicalDescr.64', '-OsvQU', 'ENTITY-MIB');
$version  = snmp_get($device, 'rndBrgVersion.0', '-OsvQU', 'RADLAN-MIB');
$serial   = snmp_get($device, 'entPhysicalSerialNum.64', '-OsvQU', 'ENTITY-MIB');

$version  = str_replace('"', '', $version);
$features = str_replace('"', '', $features);
$hardware = str_replace('"', '', $hardware);
```

I've removed $features as it made no sense since it just pointed to the backup software release. I don't know why someone did this originally.